server/tidb_test.go

```diff
@@ -1877,13 +1880,37 @@ func TestTopSQLStatementStats(t *testing.T) {
 				kvSum += kvCount
 			}
 			require.Equal(t, uint64(ExecCountPerSQL), kvSum)
+			tagChecker.checkExist(t, digest.SQLDigest, sqlStr)
 		}
 	}
 	require.Equal(t, len(sqlDigests), found)
 	require.Equal(t, 20, found)
 }
 
-func setupForTestTopSQLStatementStats(t *testing.T) (*tidbTestSuite, stmtstats.StatementStatsMap, chan struct{}, func()) {
+type resourceTagChecker struct {
+	sync.Mutex
+	sqlDigests map[stmtstats.BinaryDigest]struct{}
+}
+
+func (c *resourceTagChecker) checkExist(t *testing.T, digest stmtstats.BinaryDigest, sqlStr string) {
```

Comment:
How about checkExist for all requests happened in the test suite? In this way, when there are new kind of requests, it will be caught.

Reply:
I have a similar thought: require all requests to contain tags.

Reply:
Not all RPC requests are caused by SQL. So I can't check the all request. But maybe I can check the ALL `cop` kind  RPC request, since as I know, `cop` kind request must cause by SQL.

Reply:
Maybe we can use black lists. In this way, when there are new kind of RPC requests, people must review, check and change the test case in order to pass CI.

Reply:
I will refine this test in this issue later. https://github.com/pingcap/tidb/issues/33208